### ISSUE TITLE
[suggestion] add test name to frozen violation guid files

### ISSUE THREAD

**Issue by danhaywood** [TNG/ArchUnit#1025]
Hi, congrats on this project, really helping us in our development efforts.

We've recently started using the freezing violations feature, and while it works well, it would be useful for us if the guid files could contain, perhaps as a comment on the first line, the name of the rule that is being violated.

For example, in `stored.rules` I have:

```
classes\ that\ are\ annotated\ with\ @PersistenceCapable\ and\ not\ a\ rolled\ up\ subclass\ should\ be\ annotated\ with\ @PersistenceCapable(schema\=...)=37b32699-a3f3-42fa-b86a-22bc2d89ec18
```

and in 37b32699-a3f3-42fa-b86a-22bc2d89ec18 I have the two violations for my unit tests of the arch tests (to make sure that they are correctly detecting violations):

```
Class <org.estatio.base.test.archtests.rules.persistencecapable.pcschema.PersistenceCapableWithoutSchema> is not annotated with @PersistenceCapable(schema=...) in (PersistenceCapableWithoutSchema.java:0)
Class <org.estatio.base.test.archtests.rules.persistencecapable.pcschema.PersistenceCapableWithoutSchemaButIsRolledUpSuperclass> is not annotated with @PersistenceCapable(schema=...) in (PersistenceCapableWithoutSchemaButIsRolledUpSuperclass.java:0)
```

What I'm looking for is simply that the guid file looks something like:

```
#classes\ that\ are\ annotated\ with\ @PersistenceCapable\ and\ not\ a\ rolled\ up\ subclass\ should\ be\ annotated\ with\ @PersistenceCapable(schema\=...)
Class <org.estatio.base.test.archtests.rules.persistencecapable.pcschema.PersistenceCapableWithoutSchema> is not annotated with @PersistenceCapable(schema=...) in (PersistenceCapableWithoutSchema.java:0)
Class <org.estatio.base.test.archtests.rules.persistencecapable.pcschema.PersistenceCapableWithoutSchemaButIsRolledUpSuperclass> is not annotated with @PersistenceCapable(schema=...) in (PersistenceCapableWithoutSchemaButIsRolledUpSuperclass.java:0)
```

in other words line 1 lists the rule being violated.

**Comment by codecholeric** [comment]
Glad you like the project :slightly_smiling_face: It seems that the text-based violation store is much more "human-read" API than I originally thought :thinking: But I like your suggestion, since it's not very invasive or expensive to implement but might still make it easier to look up frozen violations for users :+1: 
Is there a reason you want all the escape characters in the comment though? If it's a comment it wouldn't be necessary to escape spaces with `\` and similar, right? And if it's about the association between the index file `stored.rules` and the details I would go via the UUID anyway, no?

**Comment by danhaywood** [comment]
My thinking was that keeping the escapes in the comment would make it easier to associate with the stored.rules file.

**Comment by codecholeric** [comment]
But for that we have the association inside of the `stored.rules` file, no? :thinking: You can even rename the file arbitrarily if you want (doesn't have to be a UUID, that's just what ArchUnit picks by default). If you already look inside of the `stored.rules` file, then why not just use the association that's there anyway?

**Comment by danhaywood** [comment]
OK thanks.  I'll close this as I have a different proposal to raise, will do so in a different ticket.
